pkg/kubectl/delete.go

```diff
@@ -148,7 +155,7 @@ func getOverlappingControllers(rcClient coreclient.ReplicationControllerInterfac
 
 func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *metav1.DeleteOptions) error {
 	rc := reaper.client.ReplicationControllers(namespace)
-	scaler := &ReplicationControllerScaler{reaper.client}
+	scaler := ScalerFor(schema.GroupKind{}, nil, reaper.scaleClient, schema.GroupResource{Resource: "replicationcontrollers"})
```

Comment:
Create a `NewScaler(scalesGetter scaleclient.ScalesGetter, gr schema.GroupResource)` function.  When scaling jobs is dead, we'll remove `ScalerFor`

Reply:
> When scaling jobs is dead, we'll remove ScalerFor

And when that dies, we'll update the `Scaler` interface to accept `GroupResource` as an arg so we don't have to choose ahead of time.